blockchain/blockchain.go

```diff
@@ -2444,6 +2444,7 @@ func (bc *BlockChain) ApplyTransaction(chainConfig *params.ChainConfig, author *
 	*usedGas += gas
 
 	receipt := types.NewReceipt(kerr.Status, tx.Hash(), gas)
+	receipt.SetCumulativeGasUsed(*usedGas)
```

Comment:
It's very minor, but it seems to appear our of thin air. 
How about adding `NewReceipt` parameter or just assign like `receipt.Logs = statedb.GetLogs(tx.Hash())` for consistency?

Reply:
Can you explain a little more? I could not understand. ;-;

Reply:
Oh, I meant that it seems to appear `SetCumulativeGasUsed` only, but not other fields. So, I suggested to add `*usedGas` while creating new receipt or just assign it `receipt.cumulativeGasUsed = *usedGas` for consistency. It is very minor, though.

Reply:
Do you meaning that passing *usedGas as the last parameter when doing `NewReceipt` and assigning it to `receipt.CumulativeGas = *usedGas` internally?
If so, early implementations did just that.
However, the test generates an error because CumulativeGasUsed is not used in other parts that use receipt, for example, the part that processes ReceiptsMsg.
In that test, the receipt test data is generated by `NewReceipt`, and because cumulativeGasUsed is added as a last parameter of the `NewReceipt`, that should use the cumulativeGasUsed also.
But when handle ReceiptsMsg, cumulativeGasUsed is not used so test will be fail because what expected (generated from NewReceipt) and got (receipt object used during handling ReceiptsMsg) is different.
This was discovered because some tests were failing, so I changed the logic to exclude CumulativeGasUsed from NewReceipt and assign it separately in this [commit](https://github.com/klaytn/klaytn/pull/1079/commits/14fd0e7011c56e179e261aa071c358ff3abf4c34).

Reply:
@jeongkyun-oh I think i misunderstood what you said, ;;; i think now i understood what you saying, and i applied in [here](https://github.com/klaytn/klaytn/pull/1079/commits/ec78c6dcf44cf30daa96f22b6fe8659d4076b0ce). Sorry for miscommunication

Reply:
Sorry for vague explanation